flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/InFlightRequestTrackerTest.java

```diff
@@ -59,4 +59,10 @@ public void testAwaitAsyncIsIdempotent() {
 		assertTrue(inFlightRequestTracker.awaitAsync().isDone());
 		assertTrue(inFlightRequestTracker.awaitAsync().isDone());
 	}
+
+	@Test
+	public void testShouldTolerateRegisterAfterAwaitAsync() {
+		assertTrue(inFlightRequestTracker.awaitAsync().isDone());
+		inFlightRequestTracker.registerRequest();
```

Comment:
I would add `assertTrue(inFlightRequestTracker.awaitAsync().isDone());` after the registration to ensure that the tracker doesn't reset or anything.

Reply:
will add